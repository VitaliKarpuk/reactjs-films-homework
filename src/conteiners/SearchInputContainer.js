import { connect } from 'react-redux';
import SearchInput from '../components/SearchInput/SearchInput';
import searchFilms from '../modules/actions/searchFilms';



const SearchInputContainer = connect(null, dispatch => ({
  onSearchFilms: (title) => {
    dispatch(searchFilms(title))
  }
}))(SearchInput);

export default SearchInputContainer;
