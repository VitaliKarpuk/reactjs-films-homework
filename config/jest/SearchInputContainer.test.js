/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import cofigureMockStore from 'redux-mock-store';

import SearchInputContainer from '../../src/conteiners/SearchInputContainer';

const mockStore = cofigureMockStore();

describe('My Connected React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      dispatch: jest.fn()
    });


    component = renderer.create(
      <Provider store={store}>
        <SearchInputContainer />
      </Provider>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with given state from Redux store', () => {
    renderer.act(() => {
      component.root.findByType('input').props.onKeyPress({ key: 'Enter' })
      expect(store.dispatch).toHaveBeenCalledTimes(1);
    });


  });
})