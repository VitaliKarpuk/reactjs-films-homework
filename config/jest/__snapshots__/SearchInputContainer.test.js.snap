// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="header__search"
>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="the jungle book"
    type="text"
    value=""
  />
  <span
    className="header__search-icon"
  />
</div>
`;
